Training24/NumConversion.c

```diff
@@ -0,0 +1,100 @@
+// ------------------------------------------------------------------------------------------------
+// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+// ------------------------------------------------------------------
+// Numconversion.c
+// Program to get user input(only integer) and convert them to Hexadecimal and Binary.
+// ------------------------------------------------------------------------------------------------
+
+#include <stdio.h>
+#include <stdbool.h>
+#include <string.h>
+#include <ctype.h>
+#include <limits.h>
+
+/// <summary>Function to  convert an integer to binary.</summary>
+void ConvertToBinary (int number, int binaryArray[], int bitWidth) {
+   for (int i = bitWidth - 1; i >= 0; i--) binaryArray[bitWidth - 1 - i] = (number >> i) & 1;
+   return;
+}
+
+/// <summary>Function to print the binary representation from the binary array.</summary>
+void PrintBinary (const int binaryArray[], int bitWidth) {
+   for (int i = 0; i < bitWidth; i++) printf ("%d", binaryArray[i]);
+   printf ("\n");
+}
+
+/// <summary>Function to  check if input is a valid integer.</summary>
+bool IsValidInteger (const char* str) {
+   if (*str == '-' || *str == '+') str++;   // Skip sign if present
+   if (!*str) return false;   // If string is empty after sign
+   while (*str) {   // Check each character to ensure it's a digit
+      if (!isdigit (*str)) return false; 
+      str++;
+   }
+   return true;
+}
+
+/// <summary>Function to get a valid integer from the user.</summary>
+bool GetValidInteger (int* number) {
+   char input[100];
+   if (fgets (input, sizeof (input), stdin) != NULL) {
+      input[strcspn (input, "\n")] = '\0';     // Remove newline character if present
+      if (IsValidInteger (input) && sscanf_s (input, "%d", number) == 1) return true;
+   }
+   return false;
+}
+
+/// <summary>Function to convert an integer into hexadecimal representation.</summary>
+void ConvertToHex (int number, char hex[], int bitWidth) {
+   int value;
+   int hexDigits = bitWidth / 4;   // Calculate number of hex digits based on bit width

```

Comment:
```suggestion
int hexDigits = bitWidth / 4;
```
You should explain what this number 4 is using comments or use a variable/sizeof, so it is not difficult in the future to change this behavior if required